This repository contains comprehensive script set for the analysis described in the M. Filippov master thesis "Predicting Genetic Variants with Discordant Post-Imputation Allele Frequency", ITMO University, 2023/2025. 
*** 
Project tested with bcftools 1.19, VEP v113, R 4.4.2. Required R packages are listed in R scripts. 
In orger to reproduce all the steps, one shall —Åcomplete folowing steps with terminal launched in repository folder:
1. Place imputed `vcf.gz` data in a repository folder. To run scripts as is, it must be named `1000G.imputed.vcf.gz`.
2. Run `QC and filtering` script. It performs QC steps, described in a thesis and generates some files needed for association analysis. 
3. Submit `1000G.yaml` generated by now and `vars_typed.txt` on SCoRe platform with default QC filters. Download results and place the whole folder (extracted) in a repository, named `controls_counts/`. Rename `counts_X.tsv` files inside this folder as `counts_typed_X.tsv`
4. Submit `1000G.yaml` generated by now and `vars_imputed.txt` on SCoRe platform with default QC filters. Download results and place the whole folder (extracted) in a repository, named `controls_counts/`. Rename `counts_X.tsv` files inside this folder as `counts_imputed_X.tsv`
5. Run `GWAS.R` script. 
Resulting  file is results_full_annotated.csv. Feel free to explore it in `exploration.Rmd`.
***

Training procedure is documented in `training.ipynb` notebook. It is not operational as we are not providing any data in this repository, but it is reproducible with any mock dataset that meets the structure described in the notebook. 
